scripts/reclaim-space.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env sh
+
+#
+# The script tries to free as much space as possible by removing nix packages
+# and docker images that aren't used.
+#
+
+set -e
+
+MIN_FREE_GB=$1
+
+get_avail_gb() {
+    echo $(( $(df --output=avail / | awk 'NR == 2 { print $1 }' ) / 1024 / 1024 ))
+}
+
+free=$(get_avail_gb)
+echo "Available space in root partition: $free GB"
+
+if [ -n "$MIN_FREE_GB" ]; then
+    if [ "$free" -gt "$MIN_FREE_GB" ]; then
+        exit 0
+    fi
+fi
+
+set -x
+docker prune --force --all
```

Comment:
Consider adding e.g. `--filter until=24h` that would remove stuff older than e.g. 24h. I'm totally oddballing here, but there might be images like busybox, base images etc... that would be re-downloaded right away anyway? So this would do some speedup costing some disk space that would most probably be used right away anyway? Maybe even different filter (longer?shorter?)

Reply:
the dockerd is used in a different way here. Basically it could be completely avoided if we used for example skopeo to upload images to the registry. The only purpose of docker is to load the archive created by nix and then push the images to registry. It does not contain any other images than mayastor images.

Reply:
I see.

Reply:
BTW, isn't better to run `docker pune --force --all` *after* the image upload then?

Reply:
good point